torchtune/training/_distributed.py

```diff
@@ -338,6 +339,8 @@ def load_from_full_model_state_dict(
                 sharded_meta_param.device_mesh,
                 sharded_meta_param.placements,
             )
+        if cpu_offload:
+            sharded_tensor = sharded_tensor.cpu()
```

Comment:
`sharded_tensor` have `device=cuda` because `distribute_tensor`/DTensor requires NCCL. For cpu offloading, we can move DTensor to `device=cpu` afterwards to avoid peaking memory